internal/dag/builder.go

```diff
@@ -514,6 +514,16 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 		secure := b.lookupSecureVirtualHost(host)
 		addRoutes(secure, routes)
 	}
+
+	// HTTPProxy object must have either a TCPProxy spec, or
+	// some combination of Include and Route specs that result
+	// routes being generated.
+	if sw.IsValid() {
```

Comment:
see comment about isValid below. I would prefer that computeRoutes return a failure case to indicate that computeRoutes encountered a failure.

Reply:
Yeh, I considered that approach but the error there would be something pretty generic like `errors.New("failed to route")`. I can revisit this though.